### PR TITLE
app-office/lyx: add version 2.3.4.2 with python3 support

### PR DESCRIPTION
**Pull request by abecerra**
Linked Bug: https://bugs.gentoo.org/695014

This PR is just lyx-2.3.3.ebuild with Laslo Hunhold patch applied:  https://bugs.gentoo.org/attachment.cgi?id=599322

I've tested with python 3.7 and Lyx works correctly.

### CODE REVIEW

File: app-office/lyx/lyx-2.3.4.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,174 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 python3_{5,6,7,8} )
```
Comment (Zlogene) [inline app-office/lyx/lyx-2.3.4.2.ebuild]: remove python2_7

**Comment by Zlogene** [comment]
also please squash 2 commits into one

File: app-office/lyx/lyx-2.3.4.2.ebuild (hunk 1 above)
Comment (abecerra) [reply app-office/lyx/lyx-2.3.4.2.ebuild]: removed

**Comment by abecerra** [comment]
> also please squash 2 commits into one

squashed

**Comment by abecerra** [comment]
The Qa bot reports problems but the link is broken

**Comment by j-licht** [comment]
A local pkgcheck scan says:
`UnnecessaryManifest: unnecessary file in Manifest: [ lyx-2.3.4.2.ebuild ]`

**Comment by abecerra** [comment]
> A local pkgcheck scan says:
> `UnnecessaryManifest: unnecessary file in Manifest: [ lyx-2.3.4.2.ebuild ]`

Thank you @j-licht

**Comment by abecerra** [comment]
@Zlogene , @j-licht: do you have other comment/suggestion ?

**Comment by Katana-Steel** [comment]
1 issues is dia requires python2.7, and it has had no updates since 2011
so to prevent python issues exclude the *dia* USE flag?

**Comment by Zlogene** [comment]
@abecerra squash commits again :)

**Comment by abecerra** [comment]
> @abecerra squash commits again :)

squashed

**Comment by Katana-Steel** [comment]
@abecerra I dont see a python3 port of app-office/dia making the diamond use flag useless

**Comment by abecerra** [comment]
> @abecerra I dont see a python3 port of app-office/dia making the dia use flag useless

shall I remove the dia flag @Katana-Steel, @Zlogene ?

**Comment by a17r** [comment]
that depends what that use flag does. so far it does not depend on any specific python implementation of dia and I do not even see it ticking a build flag, so what exactly is it doing?

File: app-office/lyx/lyx-2.3.4.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,174 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python3_{5,6,7,8} )
```
Comment (a17r) [inline app-office/lyx/lyx-2.3.4.2.ebuild]: python3_5 is long gone.

File: app-office/lyx/lyx-2.3.4.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+inherit desktop flag-o-matic font python-single-r1 qmake-utils toolchain-funcs xdg-utils
+
+DESCRIPTION="WYSIWYM frontend for LaTeX, DocBook, etc."
+HOMEPAGE="https://www.lyx.org/"
+SRC_URI="ftp://ftp.lyx.org/pub/lyx/stable/2.3.x/${MY_P}.tar.xz
+	ftp://ftp.lyx.org/pub/lyx/devel/lyx-2.3/${MY_P}/${MY_P}.tar.xz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="amd64 ~hppa ~ppc ~ppc64 ~sparc x86 ~x64-macos ~x86-macos"
```
Comment (a17r) [inline app-office/lyx/lyx-2.3.4.2.ebuild]: can't add a version bump with stable keywords, move all to unstable (`ekeyword ~all lyx-2.3.4.2.ebuild`).

File: app-office/lyx/lyx-2.3.4.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,174 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
Comment (a17r) [inline app-office/lyx/lyx-2.3.4.2.ebuild]: please bump to EAPI-7.

File: app-office/lyx/lyx-2.3.4.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+		dev-tex/html2latex
+		dev-tex/latex2rtf
+	)
+	subversion? ( dev-vcs/subversion )
+	svg? ( || ( gnome-base/librsvg media-gfx/inkscape ) )
+"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
+	dev-qt/qtconcurrent:5
+	nls? ( sys-devel/gettext )
```
Comment (a17r) [inline app-office/lyx/lyx-2.3.4.2.ebuild]: virtual/pkgconfig and sys-devel/gettext would be in `BDEPEND` in `EAPI-7`.

File: app-office/lyx/lyx-2.3.4.2.ebuild [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+		elog
+	fi
+}
+
+pkg_postrm() {
+	xdg_icon_cache_update
+	xdg_desktop_database_update
+
+	if use latex ; then
+		texhash
```
Comment (a17r) [inline app-office/lyx/lyx-2.3.4.2.ebuild]: external command needs to `|| die`

File: app-office/lyx/lyx-2.3.4.2.ebuild [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+}
+
+pkg_postinst() {
+	font_pkg_postinst
+	xdg_icon_cache_update
+	xdg_desktop_database_update
+
+	# fix for bug 91108
+	if use latex ; then
+		texhash
```
Comment (a17r) [inline app-office/lyx/lyx-2.3.4.2.ebuild]: external command needs to `|| die`

**Comment by abecerra** [comment]
> that depends what that use flag does. so far it does not depend on any specific python implementation of dia and I do not even see it ticking a build flag, so what exactly is it doing?

It seems it is there only to put dia as a dependency.

**Comment by a17r** [comment]
> It seems it is there only to put dia as a dependency.

That's another way to put my observation in words. ;) but what does the build system/readme of the package say?

**Comment by abecerra** [comment]
> > It seems it is there only to put dia as a dependency.
> 
> That's another way to put my observation in words. ;) but what does the build system/readme of the package say?

I can not find dia references in LyX sources (README, INSTALL, configure, configure.ac, CMakeLists.txt)

File: app-office/lyx/lyx-2.3.4.2.ebuild (hunk 2 above)
Comment (abecerra) [reply app-office/lyx/lyx-2.3.4.2.ebuild]: removed

File: app-office/lyx/lyx-2.3.4.2.ebuild (hunk 3 above)
Comment (abecerra) [reply app-office/lyx/lyx-2.3.4.2.ebuild]: done

File: app-office/lyx/lyx-2.3.4.2.ebuild (hunk 4 above)
Comment (abecerra) [reply app-office/lyx/lyx-2.3.4.2.ebuild]: bumped

File: app-office/lyx/lyx-2.3.4.2.ebuild (hunk 6 above)
Comment (abecerra) [reply app-office/lyx/lyx-2.3.4.2.ebuild]: added

File: app-office/lyx/lyx-2.3.4.2.ebuild (hunk 7 above)
Comment (abecerra) [reply app-office/lyx/lyx-2.3.4.2.ebuild]: added

File: app-office/lyx/lyx-2.3.4.2.ebuild [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+
+DESCRIPTION="WYSIWYM frontend for LaTeX, DocBook, etc."
+HOMEPAGE="https://www.lyx.org/"
+SRC_URI="ftp://ftp.lyx.org/pub/lyx/stable/2.3.x/${MY_P}.tar.xz
+	ftp://ftp.lyx.org/pub/lyx/devel/lyx-2.3/${MY_P}/${MY_P}.tar.xz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~hppa ~ppc ~ppc64 ~sparc ~x86 ~x64-macos ~x86-macos"
+IUSE="aspell cups debug docbook dia dot enchant gnumeric html +hunspell +latex monolithic-build nls rcs rtf subversion svg l10n_he"
```
Comment (a17r) [inline app-office/lyx/lyx-2.3.4.2.ebuild]: It seems this ebuild has some long standing problems:
- `dia` USE flag is not used except in RDEPEND
- `dot` USE flag is not used except in RDEPEND
- `cups` USE flag is not used except in RDEPEND
- `gnumeric` USE flag is not used except in RDEPEND
- `html` USE flag is not used except in RDEPEND
- `rcs` USE flag is not used except in RDEPEND, but this one I found as optional dep at least in other distro as well.
- `rtf` USE flag is not used except in RDEPEND
- `subversion` USE flag is not used except in RDEPEND
- `svg` USE flag is not used except in RDEPEND

File: app-office/lyx/lyx-2.3.4.2.ebuild [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+	dev-texlive/texlive-fontsextra
+	sys-apps/file
+	sys-libs/zlib
+	virtual/imagemagick-tools[png,svg?]
+	aspell? ( app-text/aspell )
+	cups? ( net-print/cups )
+	dia? ( app-office/dia )
+	docbook? ( app-text/sgmltools-lite )
+	dot? ( media-gfx/graphviz )
+	enchant? ( app-text/enchant:0 )
```
Comment (a17r) [inline app-office/lyx/lyx-2.3.4.2.ebuild]: Arch Linux have a patch for enchant:2 support - but before you copy it from there, please check if you can find an official upstream patch for that and get it from their repository.

Comment (abecerra) [reply app-office/lyx/lyx-2.3.4.2.ebuild]: enchant:2 is already supported I think:
https://www.lyx.org/trac/changeset/f1694db4/lyxgit

I tested changing to `enchant? ( app-text/enchant:2 )` and Lyx pulls enchant-2.2.7, merges and allows spellchecking

File: app-office/lyx/lyx-2.3.4.2.ebuild (hunk 8 above)
Comment (abecerra) [reply app-office/lyx/lyx-2.3.4.2.ebuild]: Would it make sense to remove thes useflags?

File: app-office/lyx/lyx-2.3.4.2.ebuild (hunk 9 above)
Comment (abecerra) [reply app-office/lyx/lyx-2.3.4.2.ebuild]: @a17r enchant should be changed ?

**Comment by XVilka** [comment]
Any updates on this PR? It's important to get rid of Python 2.

File: app-office/lyx/lyx-2.3.4.2.ebuild (hunk 9 above)
Comment (abecerra) [reply app-office/lyx/lyx-2.3.4.2.ebuild:41]: Changed to enchant2

File: app-office/lyx/lyx-2.3.4.2.ebuild (hunk 8 above)
Comment (abecerra) [reply app-office/lyx/lyx-2.3.4.2.ebuild:19]: Leaving for future PRs

**Comment by abecerra** [comment]
> Any updates on this PR? It's important to get rid of Python 2.

I think it can be approved now, there are newer LyX versions

File: app-office/lyx/lyx-2.3.4.2.ebuild (hunk 5 above)
Comment (abecerra) [reply app-office/lyx/lyx-2.3.4.2.ebuild]: Moved

**Comment by a17r** [comment]
@ulm, if you're interested in this package then please help get this PR over the finish line.

**Comment by ulm** [comment]
> @ulm, if you're interested in this package then please help get this PR over the finish line.

I've just bumped lyx to version 2.3.5.2. Trivial version bump, with the minimum changes necessary for python 3 support.

**Comment by ulm** [comment]
So, unless there is a reason for adding 2.3.4.2 as well, I guess we can consider this PR obsolete?

**Comment by a17r** [comment]
> So, unless there is a reason for adding 2.3.4.2 as well, I guess we can consider this PR obsolete?

This PR contains more than minimal changes, please compare. And maybe consider asking the existing maintainer to join for co-maintenance, because the linked bug was unanswered for almost a year.

**Comment by a17r** [comment]
In other words this is effectively unmaintained since https://bugs.gentoo.org/631798#c6

**Comment by ulm** [comment]
> This PR contains more than minimal changes, please compare/read comments. And maybe consider asking the existing maintainer to join for co-maintenance, because the linked bug was unanswered for almost a year.

:shrug: I was reluctant to do an EAPI bump in non-maintainer commit.

Also, looks like your comments about dropping the use-conditional dependencies from RDEPEND haven't been addressed?

**Comment by rossbridger** [comment]
Now that the latest version 2.3.5.2 is already in the tree, I think we can close this PR

**Comment by abecerra** [comment]
> Now that the latest version 2.3.5.2 is already in the tree, I think we can close this PR. If there are improvements for the ebuild, they should be applied against the latest version in a separate PR.

Closing!

**Pull request closed without merging**